src/module/posts/posts.service.ts

```diff
@@ -88,6 +88,9 @@ export class PostsService {
     }
     const user_id = user.id;
     const post = await this.getOneById(id, user_id);
+    post.comments.map(
+      comment => getRepository(Comment).remove(comment)
+    );
```

Comment:
Instead of multiple connections on DB (one for every comment), delete all coments at once.